File: Data Structures & Algorithms/top-k-elements-in-list/submission-10.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # {value : frequency} hashmap
        hashFreq = {}

        buckets = [[] for i in range(len(nums)+1)]
        ans = []
        for num in nums:
            if num in hashFreq:
                hashFreq[num] += 1
            else:
                hashFreq[num] = 1
        
        for key,value in hashFreq.items():
            buckets[value].append(key)

        print(buckets)
        print(hashFreq.items())
        for i in range(len(buckets)-1, 0, -1):
            for num in buckets[i]:
                ans.append(num)
                if len(ans) == k:
                    return ans



            

File: Data Structures & Algorithms/duplicate-integer/submission-10.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let hashmap = {}

        for(let i = 0; i < nums.length; i++){
            hashmap[nums[i]] = hashmap[nums[i]] ? hashmap[nums[i]] + 1 : 1
        }

        for(let key in hashmap){
            if(hashmap[key] > 1){
                return true
            }
        }
        return false

    }
}
